scaling_performance/network_optimization.adoc

```diff
@@ -0,0 +1,142 @@
+running on different systems.
+
+VXLAN encapsulates all tunneled traffic in user datagram protocol (UDP) packets.
+However, this leads to increased CPU utilization. Both these outer- and
+inner-packets are subject to normal checksumming rules to guarantee data has not
+been corrupted during transit. Depending on CPU performance, this additional
+processing overhead can cause a reduction in throughput and increased latency
+when compared to traditional, non-overlay networks.
+
+Cloud, VM, and bare metal CPU performance can be capable of handling much more
+than one Gbps network throughput. When using higher bandwidth links such as 10
+or 40 Gbps, reduced performance can occur. This is a known issue in VXLAN-based
+environments and is not specific to containers or {product-title}. Any network
+that relies on VXLAN tunnels will perform similarly because of the VXLAN
+implementation.
+
+If you are looking to push beyond one Gbps, you can:
+
+* Use
+xref:../install_config/configuring_routing.html#install-config-configuring-native-container-routing[Native
+Container Routing]. This option has important operational caveats that do not
+exist when using OpenShift SDN, such as updating routing tables on a router.
+* Evaluate network plug-ins that implement different routing techniques, such as
+border gateway protocol (BGP).
+* Use VXLAN-offload capable network adapters. VXLAN-offload moves the packet
+checksum calculation and associated CPU overhead off of the system CPU and onto
+dedicated hardware on the network adapter. This frees up CPU cycles for use by
+pods and applications, and allows users to utilize the full bandwidth of their
+network infrastructure.
+
+VXLAN-offload does not reduce latency. However, CPU utilization is reduced even
+in latency tests.
+
+[[scaling-performance-network-subnetting]]
+== Configuring Network Subnets
+
+{product-title} provides IP address management for both pods and services. The
+default values allow for 255 nodes, each with 254 IP addresses available for
+pods, and provides around 64,000 IP addresses for services.
+
+You can resize subnets after deployment, but this is subject to certain
+restrictions as detailed in the
+xref:../install_config/configuring_sdn.html#configuring-the-pod-network-on-masters[Configuring
+SDN documentation].
+
+On installation, you can add the following to the `[OSE3:vars]` section in your
+Ansible inventory file:
+
+----
+[OSE3:vars]
+# a /10 with osm_host_subnet_length=9 means we can have 32768 nodes.  obviously we dont support anywhere near that.
+osm_cluster_network_cidr=172.20.0.0/14
+
+# a /10 gives us 4 million usable IPs for services.  this number is wildly oversized.  we just do it to keep the IP numbering simple
+openshift_portal_net=10.192.0.0/10
+
+# The size of the per host subnet allocated for pod IPs
+# 9 gives us a /23 on each node, ~510 usable
+osm_host_subnet_length=8
+----
+
+This will allow for around 1000 nodes, each with approximately 250 IPs available
```

Comment:
Forget about portal_net because it doesn't relate to this node/pod density sizing section.  portal_net is just a pool of IPs for services.

```
osm_cluster_network_cidr=172.20.0.0/14

$ whatmask 172.20.0.0/14

------------------------------------------------
           TCP/IP NETWORK INFORMATION
------------------------------------------------
IP Entered = ..................: 172.20.0.0
CIDR = ........................: /14
Netmask = .....................: 255.252.0.0
Netmask (hex) = ...............: 0xfffc0000
Wildcard Bits = ...............: 0.3.255.255
------------------------------------------------
Network Address = .............: 172.20.0.0
Broadcast Address = ...........: 172.23.255.255
Usable IP Addresses = .........: 262,142
First Usable IP Address = .....: 172.20.0.1
Last Usable IP Address = ......: 172.23.255.254
```

Start with the Network Address and Broadcast Address.

If ```osm_host_subnet_length=8``` then 
The first node will be given 172.20.0.0/24.
The second node will be given 172.20.1.0/24
...
Up through 172.23.255.0/24 which is the 1000th node.

Each node has a /24 allocated to it. The first and last IPs are unusable, leaving 254 usable per node.  We state a max-pods of 250 supported.

Does this help?